R/edgeThresh.R

```diff
@@ -0,0 +1,284 @@
+#'     the components of the graph is measured initially, and the component
+#'     each node belongs to is stored in a separate data structure (hash table?).
+#'     After each swap, the new component of the sending endpoint (A in A → B)
+#'     is the same as the component the receiving endpoint belongs to. Thus
+#'     the component maintenance is acieved in 2 steps (one query and one
+#'     comparison) for each receiving node, and thus 4 * Q * |E| steps are
+#'     required.
+#'
+#'     ## Why not use a binary search style approach to find the minimum delta?
+#'     ## Can calculation of delta be parallelized?
+#' @section Incrementing δ and Lmax: ## How much should delta be incremented for efficieny as well as accuracy?##
+#'     After creating the permutated graph, the following procedure is run on
+#'     the graph for each Lmax starting from 2 to the Lmax given as an argument.
+#'     On each such run, and a small value for δ is chosen (0.01) , and
+#'     the strongly connected component discovery is done. If the size of the
+#'     biggest component after such run is smaller than Lmax, increment δ by
+#'     (0.01), and redo component disovery. As soon as the largest component has
+#'     size Lmax (for current run), δ is recorded, and Lmax is incremented by 1.
+#'     Once current Lmax is the same as the given Lmax, run the procedure on the
+#'     next graph.
+#'     The above procedure is done on each of the N graphs, Lmax - 1 times.
+#'
+#'
+#'
+#'
+#' @param EGG An EGG object - the input heat-equilibrated gene score graph.
+#' @param Lmax	Maximum order for largest connected component in a random
+#' network. See details.
+#' @param N	The number of permuted networks to produce for choosing a δ
+#' cutoff.
+#' @param Q	The number of edge-swapping permutations to produce one of the
+#' permuted networks is Q * |E|.
+#' @param silent Boolean. Default: FALSE. Whether or not to write
+#' progress information to console.
+#' @param writeLog Boolean. Default: TRUE Whether or not to log results.
+#'
+#' @return The Equilibrated Gene Graph EGG with δ values as additional
+#' graph metadata.
+#'
+#' @author Farzan Taj farzantaj@outlook.com
+#' @references
+#' @seealso
+#' @examples
+#'
+#' @export
+THRESH <- function(EGG,
+                   Lmax = max(getOptions("rete.minSubnetOrder")),
+                   N = 100,
+                   Q = 100,
+                   silent = FALSE,
+                   writeLog = TRUE) {
+    # NOTE: Most comments, only as a confirmation/question
+
+
+    # Check arguments for validity, generate error if necessary and stop
+    # Read EGG
+    # Use the helper function ".permute" to create permutations,
+    # which takes a copy of EGG, and N, and returns N random
+    # graphs as a list (?) of objects
+
+
+    # If variables are assigned in the local environmet, the helper functions
+    # can use them.
+    # Generate the list of edges, vertices and vertex count available in EGG
```

Comment:
???? Just pass an EGG graph to the helper ... it's going to be only read-from anyway.